app/Http/Controllers/Dashboard/DashboardController.php

```diff
@@ -79,9 +79,26 @@ public function showDashboard()
         $components = Component::orderBy('order')->get();
         $incidents = $this->getIncidents();
         $subscribers = $this->getSubscribers();
-        $usedComponentGroups = Component::enabled()->where('group_id', '>', 0)->groupBy('group_id')->pluck('group_id');
-        $componentGroups = ComponentGroup::whereIn('id', $usedComponentGroups)->orderBy('order')->get();
-        $ungroupedComponents = Component::enabled()->where('group_id', 0)->orderBy('order')->orderBy('created_at')->get();
+
+        // Component & Component Group lists.
+        $usedComponentGroups = Component::enabled()->where('group_id', '>', 0)
+            ->groupBy('group_id')
+            ->pluck('group_id')
+        ;
+        $componentGroupsBuilder = ComponentGroup::guest();
+        if (auth()->check()) {
+            $componentGroupsBuilder = ComponentGroup::loggedIn(auth()->user());
+        }
+        $componentGroups = $componentGroupsBuilder->whereIn('id', $usedComponentGroups)
+            ->orderBy('order')
+            ->get()
+        ;
+        $ungroupedComponents = Component::enabled()
+            ->where('group_id', 0)
+            ->orderBy('order')
+            ->orderBy('created_at')
+            ->get()
+        ;
```

Comment:
The reasons I chose to do it like this are:
- easier to debug by commenting out one line(or more)
- easier to delete the last line if you don't need it anymore, without having to do the following operations: delete the line with `;` at the end, move to the previous line, go to the end of the line, add the ';'
- easier to add a new line(e.g. call to another method) after the last line, without having to do the following operations: delete the `;` from the last line, a the new line, add the `;` to the new last line

Also, IMHO, it's more readable writing each method in a chain call on its own line.